misk/src/main/kotlin/misk/web/jetty/Socket.kt

```diff
@@ -0,0 +1,30 @@
+package misk.web.jetty
+
+import misk.logging.getLogger
+import org.eclipse.jetty.websocket.api.Session
+import org.eclipse.jetty.websocket.api.WebSocketAdapter
+import javax.inject.Inject
+
+private val logger = getLogger<Socket>()
+
+class Socket(private val wsConnections: WsConnections) : WebSocketAdapter() {
+    override fun onWebSocketConnect(session: Session) {
+        super.onWebSocketConnect(session)
+        logger.debug { "Socket connected: [session=$session]" }
+        wsConnections.subscribe(this, "all")
+    }
+
+    override fun onWebSocketClose(
+            statusCode: Int,
+            reason: String?
```

Comment:
Is reason actually nullable? Often Intellij does the wrong thing with nullability when creating overrides for methods defined in Java